var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -50,6 +58,9 @@ def cmake_args(self):
         else:
             options.append('-DENABLE_CUDA=Off')
 
+        if '+c' in spec:
+            options.append('-DENABLE_C=On')
```

Comment:
Can you explicitly disable these in an else-clause? CMake likes to enable things based on whether or not compilers/dependencies are found, leading to non-deterministic non-reproducible builds.

Reply:
In Umpire, turning on ENABLE_FORTAN (for now) also turns on ENABLE_C:
```if (ENABLE_FORTRAN)
  set(ENABLE_C On)
endif()
```
I'm not sure I want to override the software on this point.  Umpire uses BLT, so the options are:
* Turn on ENABLE_FORTRAN, which turns on ENABLE_C, which requires fortran and C compilers
* Turn on ENABLE_C, which requires just C compiler.

According to @davidbeckingsale , building the C interface without requiring Fortran is a valid build configuration, especially when using clang compilers.

What's the proper resolution here? 
* Do I encode this dependency in the package?  
* Do I rely on the CMake system that underlies everything to figure this out?
* What happens if the CMake system that underlies this changes? Do I recode the package?


Also, should I do this for `numa` as well?

Reply:
I should also add-- the `ENABLE_C` option is just there to tell BLT that a C compiler is required.  It's not turned on by CMake.

Reply:
In that case, ignore my above suggestion and add:
```python
conflicts('~c', when='+fortran', msg='Fortran API requires C API')
```
This prevents users from selecting an invalid configuration (`~c+fortran`).